src/tasks.js

```diff
@@ -1,19 +1,23 @@
 const generateId = () => (Math.random() + 1).toString(36).substring(7);
 
-export const Task = (description) => {
-  const completedStatus = false;
-  const id = generateId();
+export const Task = (taskData) => {
+  const { description, completedStatus = false, id = generateId() } = taskData;
   function changeCompletedStatus() {
     this.completedStatus = !this.completedStatus;
   }
+  function setDescription(newDescription) {
+    this.description = newDescription;
+  }
   return {
-    description, completedStatus, id, changeCompletedStatus,
+    description, completedStatus, id, changeCompletedStatus, setDescription,
   };
 };
 
-const taskOne = Task('Go groceries shopping');
-const taskTwo = Task('Wash the plates');
-const taskThree = Task('Do my laundry');
+const parseTasks = (tasks) => tasks.map(Task);
+
+const taskOne = Task({ description: 'Go groceries shopping' });
+const taskTwo = Task({ description: 'Wash the plates' });
+const taskThree = Task({ description: 'Do my laundry' });
```

Comment:
- [ ] Please remove all hardcoded tasks from your code. Initially the tasks array is supposed to be empty.

Reply:
Thanks, this has been resolved